include/envoy/network/connection_handler.h

```diff
@@ -68,9 +68,51 @@ class ConnectionHandler {
    * after they have been temporarily disabled.
    */
   virtual void enableListeners() PURE;
+
+  /**
+   * Used by ConnectionHandler to manage listeners.
+   */
+  class ActiveListener {
+  public:
+    virtual ~ActiveListener() = default;
+
+    /**
+     * @return the tag value as configured.
+     */
+    virtual uint64_t listenerTag() PURE;
+    /**
+     * @return the actual Listener object.
+     */
+    virtual ListenerPtr& listener() PURE;
+  };
+
+  using ActiveListenerPtr = std::unique_ptr<ActiveListener>;
 };
 
 using ConnectionHandlerPtr = std::unique_ptr<ConnectionHandler>;
 
+/**
+ * A registered factory interface to create different kinds of
+ * ActiveUdpListener.
+ */
+class ActiveUdpListenerFactory {
+public:
+  virtual ~ActiveUdpListenerFactory() = default;
+
+  /**
+   * Creates an ActiveUdpListener object and a corresponding UdpListener
+   * according to given config.
+   * @param parent is the owner of the created ActiveListener objects.
+   * @param config provides information needed to create ActiveUdpListener and
+   * UdpListener objects.
+   * @return the ActiveUdpListener created.
+   */
+  virtual ConnectionHandler::ActiveListenerPtr
+  createActiveUdpListener(ConnectionHandler& parent, Event::Dispatcher& disptacher,
+                          spdlog::logger& logger, Network::ListenerConfig& config) const PURE;
```

Comment:
nit: I would prefer to not pass the logger through the interface. Can we just skip this and use a statically defined logger inside the listener class? If you want to deal with this in a follow up feel free to add a TODO.

Reply:
Right now, conn_handler's logger is passed in instead of defined statically in the class. Though the logger is eventually ENVOY_LOGGER(), I don't know why it is so, maybe for testing purpose? Added a TODO to investigate later.